src/storage/cache/rpc/packet.rs

```diff
@@ -0,0 +1,524 @@
+            return Err(RpcError::InvalidResponse(
+                "Invalid response header".to_owned(),
+            ));
+        }
+
+        let seq = get_u64_from_buf(buf, 0)?;
+        let op = get_u8_from_buf(buf, 8)?;
+        let len = get_u64_from_buf(buf, 9)?;
+
+        Ok(RespHeader { seq, op, len })
+    }
+}
+
+/// Define the Packet trait, for once send or receive packets
+///
+/// Client will create a packet and serialize it to a byte array,
+/// send it to the server. And create a temp response packet in client.
+///
+/// Server will receive the packet, deserialize it to a packet struct,
+/// and process the request, then create a response packet and serialize.
+///
+/// Client will receive the response packet and deserialize it to a packet struct.
+/// and check the status of the packet and the response.
+pub trait Packet: Sync + Send + Clone + Debug {
+    /// Get the packet seq number
+    fn seq(&self) -> u64;
+    /// Set the packet seq number
+    fn set_seq(&mut self, seq: u64);
+
+    /// Get packet type
+    fn op(&self) -> u8;
+    /// Set packet type
+    fn set_op(&mut self, op: u8);
+
+    /// Serialize request data to bytes
+    fn set_req_data(&mut self, data: &[u8]) -> Result<(), RpcError<String>>;
+
+    /// Get the serialized request data
+    fn get_req_data(&self) -> Result<Vec<u8>, RpcError<String>>;
+
+    /// Serialize response data to bytes
+    fn set_resp_data(&mut self, data: &[u8]) -> Result<(), RpcError<String>>;
+
+    /// Deserialize response data from bytes
+    fn get_resp_data(&self) -> Result<Vec<u8>, RpcError<String>>;
+
+    /// Get the packet status
+    fn status(&self) -> PacketStatus;
+
+    /// Set the packet status
+    fn set_status(&mut self, status: PacketStatus);
+}
+
+/// The `PacketStatus` enum is used to represent the status of a packet.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum PacketStatus {
+    /// The packet is pending.
+    Pending,
+    /// The packet is successful.
+    Success,
+    /// The packet is failed.
+    Failed,
+    /// The packet is timeout.
+    Timeout,
+}
+
+impl PacketStatus {
+    /// Convert the `PacketStatus` to u8
+    #[must_use]
+    pub fn to_u8(&self) -> u8 {
+        match *self {
+            PacketStatus::Pending => 0,
+            PacketStatus::Success => 1,
+            PacketStatus::Failed => 2,
+            PacketStatus::Timeout => 3,
+        }
+    }
+
+    /// Convert u8 to `PacketStatus`
+    #[must_use]
+    pub fn from_u8(status: u8) -> Self {
+        match status {
+            0 => PacketStatus::Pending,
+            1 => PacketStatus::Success,
+            3 => PacketStatus::Timeout,
+            _ => PacketStatus::Failed,
+        }
+    }
+}
+
+/// The `PacketsInner` struct is used to store the current tasks.
+#[derive(Debug)]
+struct PacketsInner<P>
+where
+    P: Packet + Send + Sync,
+{
+    /// current tasks, marked by the seq number
+    packets: HashMap<u64, P>,
```

Comment:
These two hashmap can be merged into one to reduce operation cost

Reply:
Ok.

Reply:
I've add a `set_timestamp` for `Packet`, so we don't need another map.